xc7/fasm2bels/ioi_models.py

```diff
@@ -0,0 +1,103 @@
+from .verilog_modeling import Site  # , Bel
+
+
+def get_ioi_site(db, grid, tile, site):
+    """
+    Returns a prxjray.tile.Site object for given ILOGIC/OLOGIC/IDELAY site.
+    """
+
+    gridinfo = grid.gridinfo_at_tilename(tile)
+    tile_type = db.get_tile_type(gridinfo.tile_type)
+
+    site_type, site_y = site.split("_")
+
+    sites = tile_type.get_instance_sites(gridinfo)
+    sites = [s for s in sites if site_type in s.name]
+    sites.sort(key=lambda s: s.y)
+
+    if len(sites) == 1:
+        iob_site = sites[0]
+    else:
+        iob_site = sites[1 - int(site[-1])]
+
+    return iob_site
+
+
+def process_idelay(top, features):
+
+    aparts = features[0].feature.split('.')
+    # tile_name = aparts[0]
+    ioi_site = get_ioi_site(top.db, top.grid, aparts[0], aparts[1])
+
+    site = Site(features, ioi_site)
+
+    # TODO: Support IDELAY
+    assert not site.has_feature("IN_USE")
+
+
+def process_ilogic(top, features):
+
+    aparts = features[0].feature.split('.')
+    # tile_name = aparts[0]
+    ioi_site = get_ioi_site(top.db, top.grid, aparts[0], aparts[1])
+
+    site = Site(features, ioi_site)
+
+    # TODO: Support IDDR, ISERDES etc
```

Comment:
Same story as IDELAY, please add assert that ISERDES/IDDR IN_USE features are not set.  This applies for the OLOGIC block too.

Reply:
I added asserts for ISERDES, IDELAY and OSERDES.